src/industry_gui.cpp

```diff
@@ -1401,8 +1414,6 @@ class IndustryDirectoryWindow : public Window {
 		this->industries.Filter(filter);
 		this->industries.Sort();
 
-		this->vscroll->SetCount((uint)this->industries.size()); // Update scrollbar as well.
```

Comment:
I am not sure this line should be moved as I think the `Filter()` function can shorten the list so the scrollbar update needed here too.